accepted/2021/statics-in-interfaces/README.md

```diff
@@ -789,90 +814,129 @@ namespace System
         static abstract TSelf TrailingZeroCount(TSelf value);
     }
 
-    public interface IFloatingPoint<TSelf>
-        : ISignedNumber<TSelf>
-        where TSelf : IFloatingPoint<TSelf>
+    public interface IExponentialFunctions<TSelf>
+        where TSelf : IExponentialFunctions<TSelf>
     {
-        // This currently implies IEEE floating-point types and so decimal does not implement it
-        // If we want decimal to implement it, we would need to move several methods down and define
-        // some IeeeFloatingPoint interface instead. This wasn't done for simplicity and because decimal
-        // is not an industry standard type.
+        static abstract TSelf Exp(TSelf x);
 
-        // TODO: The Exponent and Signifcand need to be exposed here
-        // They should return INumber<TSelf> here and IBinaryNumber<TSelf> on IBinaryFloatingPoint<TSelf>
+        // IEEE defines n to be an integral type, but not the size
 
-        // TODO: We may want to expose constants related to the subnormal and finite boundaries
+        static abstract TSelf ScaleB<TInteger>(TSelf x, TInteger n)
+            where TInteger : IBinaryInteger<TInteger>;
 
-        // The following constants are defined by System.Math and System.MathF today
+        // The following methods are approved but not yet implemented in the libraries
 
-        static abstract TSelf E { get; }
+        static abstract TSelf ExpM1(TSelf x);
 
-        static abstract TSelf PI { get; }
+        static abstract TSelf Exp2(TSelf x);
 
-        static abstract TSelf Tau { get; }
+        static abstract TSelf Exp2M1(TSelf x);
 
-        // The following methods are defined by System.Math and System.MathF today
-        // Exposing them on the interfaces means they will become available as statics on the primitive types
-        // This, in a way, obsoletes Math/MathF and brings float/double inline with how non-primitive types support similar functionality
-        // API review will need to determine if we'll want to continue adding APIs to Math/MathF in the future
+        static abstract TSelf Exp10(TSelf x);
 
-        static abstract TSelf Acos(TSelf x);
+        static abstract TSelf Exp10M1(TSelf x);
+    }
 
+    public interface IHyperbolicFunctions<TSelf>
+        where TSelf : IHyperbolicFunctions<TSelf>
+    {
         static abstract TSelf Acosh(TSelf x);
 
-        static abstract TSelf Asin(TSelf x);
-
         static abstract TSelf Asinh(TSelf x);
 
-        static abstract TSelf Atan(TSelf x);
+        static abstract TSelf Atanh(TSelf x);
 
-        static abstract TSelf Atan2(TSelf y, TSelf x);
+        static abstract TSelf Cosh(TSelf x);
 
-        static abstract TSelf Atanh(TSelf x);
+        static abstract TSelf Sinh(TSelf x);
 
-        static abstract TSelf BitIncrement(TSelf x);
+        static abstract TSelf Tanh(TSelf x);
+    }
 
-        static abstract TSelf BitDecrement(TSelf x);
+    public interface ILogarithmicFunctions<TSelf>
```

Comment:
In these `I*Functions` interfaces, there is no relationship between `ILogarithmicFunctions` and `IExponentialFunctions` or `IPowerFunctions` and `IRootFunctions`, even though they are "inverses".

We could group them together, but that would force users to "implement more" and we would need better names.